hyper/http11/connection.py

```diff
@@ -75,6 +79,21 @@ def __init__(self, host, port=None, secure=None, ssl_context=None,
         self.ssl_context = ssl_context
         self._sock = None
 
+        # Setup proxy details if applicable.
+        if proxy_host:
+            if proxy_port is None:
+                try:
+                    self.proxy_host, self.proxy_port = proxy_host.split(':')
```

Comment:
Hmm, if everything else is wrong too then I'd skip it for now. Let's open a GitHub issue to track it instead: see #170.